### ISSUE TITLE
Integration tests

### ISSUE THREAD

**Issue by texastoland** [texastoland/nabs#5]

**Comment by artlogic** [comment]
I think this is a good idea. I really like Jest's "mock-by-default" philosophy.

**Comment by texastoland** [comment]
Yep and developer experience is super polished compared to testdouble. Its stack traces were impossible. My goal is to not need mocks though. #3 ensures all state can be passed as arguments (as opposed to hidden in instance variables).

I saw snapshot tests for the first time fixing something in ESLint. You pretty print the result once then verify you keep getting the same thing. For unit tests we could either do [jest-in-case](https://github.com/atlassian/jest-in-case) or [property](https://yarnpkg.com/en/package/testcheck) [testing](https://yarnpkg.com/en/package/jsverify).
